File: Exercise/Medium/01. Palindrome Check/Code/Palindrome Check.py
def palindrome_check(line):
    for i in range(len(line) // 2):
        if line[i] != line[-i - 1]:
            return False
    return True

if __name__ == "__main__":
    line = input()

    if palindrome_check(line):
        print("True")
    else:
        print("False")

File: Exercise/Medium/01. Palindrome Check/Code/Palindrome Check.java
import java.util.Scanner;

public class Main {
    static boolean palindromeCheck(String line) {
        for (int i = 0; i < line.length() / 2; i++) {
            if (line.charAt(i) != line.charAt(line.length() - i - 1)) {
                return false;
            }
        }
        return true;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String line = scanner.nextLine();

        if (palindromeCheck(line)) {
            System.out.println("True");
        } else {
            System.out.println("False");
        }
    }
}



